lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -571,12 +571,17 @@ void TypeLoweringVisitor::visitDecl(RegOp op) {
 
   // Loop over the leaf aggregates.
   for (auto field : fieldTypes) {
-    SmallString<16> loweredName(op.nameAttr().getValue());
-    loweredName += field.suffix;
+    SmallString<16> loweredName("");
```

Comment:
How about structuring this as:

```
 StringAttr loweredName;
 if (auto nameAttr = op.nameAttr())
      loweredName = builder->getStringAttr(nameAttr.getValue().str+field.suffix);
  ```

that will eliminate the extra check in the create call.

Reply:
Thanks, simplified it.